Marlin/src/inc/SanityCheck.h

```diff
@@ -2193,13 +2193,13 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
 #if ENABLED(USE_CONTROLLER_FAN)
   #if !HAS_CONTROLLER_FAN
     #error "USE_CONTROLLER_FAN requires a CONTROLLER_FAN_PIN. Define in Configuration_adv.h."
-  #elif E0_AUTO_FAN_PIN == CONTROLLER_FAN_PIN
+  #elif PIN_EXISTS(E0_AUTO_FAN) && E0_AUTO_FAN_PIN == CONTROLLER_FAN_PIN
```

Comment:
Is there some way that `CONTROLLER_FAN_PIN` and `E0_AUTO_FAN_PIN` could both be zero? Pin 0 is usually reserved as a serial port pin.

Since `HAS_CONTROLLER_FAN` must be enabled for these tests to proceed, and `HAS_CONTROLLER_FAN` is set because `PIN_EXISTS(CONTROLLER_FAN)` there is no need to worry about both `Ex_AUTO_FAN_PIN ` and `CONTROLLER_FAN_PIN` to ever both be equal to -1.

Reply:
Looks like `PA1` resolves to `A1` which apparently is zero. Meanwhile it's conditional #undef of `Ex_AUTO_FAN_PIN` leading those to look like 0 to the preprocessor. So, looks like we have a necessary patch here!